src/main/java/org/rumbledb/compiler/RuntimeIteratorVisitor.java

```diff
@@ -941,28 +940,19 @@ public RuntimeIterator visitSwitchExpression(SwitchExpression expression, Runtim
 
     @Override
     public RuntimeIterator visitTypeSwitchExpression(TypeSwitchExpression expression, RuntimeIterator argument) {
-        List<TypeSwitchCase> cases = new ArrayList<>();
-        for (TypeSwitchCaseExpression caseExpression : expression.getCases()) {
-            String caseVariableName = null;
-            if (caseExpression.getVariableReference() != null) {
-                caseVariableName = caseExpression.getVariableReference().getVariableName();
-            }
+        List<TypeSwitchRuntimeIteratorCase> cases = new ArrayList<>();
+        for (TypeswitchCase caseExpression : expression.getCases()) {
```

Comment:
I really like that you have removed the unnecessary complexity with the variable name usage.

Reply:
Happy to hear!